### PR TITLE
fix: media library multiple selection

### PR DESCRIPTION
**Pull request by damuso**
### What does it do?

Fixed bugs:

- missing condition for `select all` button highlight in media library
- allow to select all files from multiple folders/pages without removing already selected files 

### Why is it needed?

To provide better user experience when selecting multiple files for fields with multiple media type

### Related issue(s)/PR(s)

[#16801](https://github.com/strapi/strapi/issues/16801)

### CODE REVIEW

File: packages/core/helper-plugin/src/hooks/useSelectionState/index.js [hunk 1]
```diff
@@ -34,5 +34,31 @@ export const useSelectionState = (keys, initialValue) => {
...
+  const deselectMultiple = (nextSelections) => {
+    setSelections([
+      ...selections.filter(
+        (currentSelection) =>
+          nextSelections.findIndex((nextSelection) =>
+            keys.every((key) => currentSelection[key] === nextSelection[key])
+          ) === -1
+      ),
+    ]);
+  };
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/hooks/useSelectionState/index.js:58]: I find this code hard to read at the moment, can we please add notes to explain at a high level what we're trying to do with an example?

File: packages/core/helper-plugin/src/hooks/useSelectionState/index.js [hunk 2]
```diff
@@ -34,5 +34,31 @@ export const useSelectionState = (keys, initialValue) => {
...
+        (nextSelection) =>
+          selections.findIndex((currentSelection) =>
+            keys.every((key) => currentSelection[key] === nextSelection[key])
+          ) === -1
+      ),
+    ]);
+  };
+
+  const deselectMultiple = (nextSelections) => {
+    setSelections([
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/hooks/useSelectionState/index.js:50]: I would use the function format of `setSelections` – `setSelections(currSelections => ...` this is more inline with the rest of the hook.

File: packages/core/helper-plugin/src/hooks/useSelectionState/index.js
Comment (joshuaellis) [inline packages/core/helper-plugin/src/hooks/useSelectionState/index.js:1]: You're new API changes need tests please to illustrate the behaviour clearly and so we can ensure there are no regressions in the future.

**Review by joshuaellis — CHANGES_REQUESTED** [review]
Thanks for the PR, sorry about the wait. Lets refine the `useSelectionState` additions then i'll understand the changes in `BrowseStep` better :)

File: packages/core/helper-plugin/src/hooks/useSelectionState/index.js (hunk 1 above)
Comment (damuso) [reply packages/core/helper-plugin/src/hooks/useSelectionState/index.js]: I have added comments to the code for clarity. In summary, these two methods are responsible for handling the selection of multiple files without modifying the already selected items. Additionally, they should prevent duplicate entries in the array of selected items. Initially, I considered using the some method to check if an object exists in the array, but ultimately opted for findIndex to maintain consistency within this hook.

File: packages/core/helper-plugin/src/hooks/useSelectionState/index.js (hunk 2 above)
Comment (damuso) [reply packages/core/helper-plugin/src/hooks/useSelectionState/index.js]: Thanks for the advice. I absolutely forgot about that possibility and it definitely makes it much more readable

File: packages/core/helper-plugin/src/hooks/useSelectionState/index.js
Comment (damuso) [reply packages/core/helper-plugin/src/hooks/useSelectionState/index.js:1]: I have added tests for new methods and also some description in Storybook

**Review by joshuaellis — APPROVED** [review]
Tested 👍🏼

**Pull request merged**